K_Way_Merge_Kafka/KWayMerge/src/main/java/com/example/kafka/KafkaMergeConsumer.java

```diff
@@ -0,0 +1,70 @@
+package com.example.kafka;
+
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.common.protocol.types.Field;
+import org.apache.kafka.common.serialization.StringDeserializer;
+
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Properties;
+
+public class KafkaMergeConsumer {
+    int PARTITION_COUNT;
+    Consumer<String, String> consumer;
+
+    ArrayList<Integer>[] perPartitionData;
+
+    public KafkaMergeConsumer(String bootstrapServer , int partitionCount) {
+        this.PARTITION_COUNT = partitionCount;
+        Properties consumerProps = new Properties();
+        consumerProps.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServer); // Replace with your Kafka broker addresses
+        consumerProps.put(ConsumerConfig.GROUP_ID_CONFIG, "test_group_id"); // Consumer group ID
+        consumerProps.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class.getName());
+        consumerProps.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class.getName());
+        this.consumer = new KafkaConsumer<>(consumerProps);
+        this.consumer.subscribe(Collections.singletonList("test_topic")); // Replace with your topic name
+        this.perPartitionData = new ArrayList[PARTITION_COUNT];
+
+        for (int i = 0; i < perPartitionData.length; i++) {
+            perPartitionData[i] = new ArrayList<>();
```

Comment:
I'm open to discussion on this but generally we should use a performant, thread-safe queue. One example could be this[https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ConcurrentLinkedQueue.html](url)

Reply:
Thats what I ended up using take a look